common/metrics/otel_metrics_handler.go

```diff
@@ -37,13 +38,36 @@ import (
 )
 
 // otelMetricsHandler is an adapter around an OpenTelemetry [metric.Meter] that implements the [Handler] interface.
-type otelMetricsHandler struct {
-	l           log.Logger
-	tags        []Tag
-	provider    OpenTelemetryProvider
-	excludeTags excludeTags
-	catalog     catalog
-}
+type (
+	otelMetricsHandler struct {
+		l            log.Logger
+		set          attribute.Set
+		provider     OpenTelemetryProvider
+		tagConverter tagConverter
+		catalog      catalog
+		gauges       *sync.Map // string -> *gaugeAdapter. note: shared between multiple otelMetricsHandlers
+	}
+
+	tagConverter func(Tag) attribute.KeyValue
+
+	// This is to work around the lack of synchronous gauge:
+	// https://github.com/open-telemetry/opentelemetry-specification/issues/2318
```

Comment:
It took me a long time to read this issue--I think we can summarize a bit here instead of just linking. The term "synchronous gauge" isn't immediately clear. I think the main point is that otel only natively provides methods to add / subtract from a gauge, or set its value in a callback, so we provide an adapter here to make it more ergonomic to immediately set the gauge value.

Reply:
I wasn't intending that anyone read the whole long conversation, just linking to the main discussion for reference. I can elaborate a little more